### DISCUSSION TITLE
Getting started instructions are missing quite a few steps

### DISCUSSION THREAD

**Discussion by intellix** [scullyio/scully#1517] [General]
Was just looking at Scully as an alternative to Universal and figured I'd try getting started:

```bash
ng new blah
cd blah
ng add @scullyio/init // I chose the first option "Scully platform server" (not puppeteer), as it's the first/default
ng build --prod
npx scully
```

This gives me the following error:
```
➜  blah git:(master) ✗ npx scully
  ✔ Folder "./scully" used for custom plugins
  x  Notice:
      ============================================================
       The scully-plugin-puppeteer is not installed. please run:
         npm install @scullyio/scully-plugin-puppeteer
       and try again.
      ============================================================
```

I can see that it's generated a config with a reference to an "spsModulePath":
```
spsModulePath: 'YOUR OWN MODULE PATH HERE',
```

After a google, I find this (no mention of it in the getting started guide): https://github.com/scullyio/scully/blob/dc0ae5d0db220a41c90808c69d968f8357e27ace/scully.sps-sample.config.ts#L21

I can also see this line missing in my generated config, which is in the sample one I randomly found, which now complains that I need my module... so I copy the one in this repository and add that missing command
```
enableSPS();
```

Next error is this:
```
⠙ Rendering using Scully Platform Server Error: No ResourceLoader implementation has been provided. Can't read the url "app.component.html"
```

Not seeing any reference to SPS inside the Getting Started guide. Got a feeling it's all geared towards using puppeteer? https://scully.io/docs/learn/getting-started/overview/

I then tried again with puppeteer (a quick read suggests this was the default beforehand) to see if the process was smoother and generating I see something about NaN seconds: 

```
➜  blah git:(master) ✗ npx scully
  ✔ Folder "./scully" used for custom plugins
  ✔ new Angular build files imported
  ✔ Starting servers for project "blah"
  ✔ Started Scully static server on "http://localhost:1668/"
  ✔ Started Angular distribution server on "http://localhost:1864/" 
  ✔ Scully Development Server is up and running
  ⚠ (You can skip this, or any future question by using the --noPrompt flag)
Would you allow Scully to collect anonymous errors to improve our services? (Y/n)
  ✔ Puppeteer is being launched
  ✔ Successfully scanned Angular app for routes
  ✔ Successfully added routes created from routePlugins
  ✔ Route list created in files:
     "./src/assets/scully-routes.json",
     "/Users/dominicwatson/Projects/gitlab/the-mill/blah/dist/static/assets/scully-routes.json",
     "/Users/dominicwatson/Projects/gitlab/the-mill/blah/dist/blah/assets/scully-routes.json"

  ✔ Route "/" rendered into "./dist/static/index.html" 

Total time used 35.47 seconds
  1 pages have been created
  Rendering the pages took NaN seconds
  That is NaN pages per second,
  or NaN milliseconds for each page.
  
  Pulling in route-data took NaN seconds
```

And serving seems not to work out of the box:
```
➜  blah git:(master) ✗ npm run scully:serve

> blah@0.0.0 scully:serve /Users/dominicwatson/Projects/gitlab/the-mill/blah
> npx scully serve --

  ✔ Starting servers for project "blah"
  ✔ Scully development Servers stopped, and exited
  ✔ Scully development Servers stopped, and exited
```

**Comment by intellix** [comment]
I've found it in the docs by looking at the Releases and seeing SPS mentioned in 2.1.0. PR where documentation was added here: https://github.com/scullyio/scully/pull/1478/files 

But it's hidden inside Reference -> RouteRender Plugins -> SPS Scully Platform Server: https://scully.io/docs/Reference/routRenders/sps/

**Comment by SanderElias** [comment]
@intellix I think we fixed those things now with updates in our schematics, and the getting started should be the same as it was now.
There was some turmoil when we moved the puppeteerRenderer outside of Scully's core, and into its own package.
The SPS renderer is faster, and doesn't use anything outside of Angular, but isn't suited for all projects (just yet)

Sorry for the late reply, this one escaped my attention somehow.
